src/values/instruction_value.rs

```diff
@@ -203,25 +213,49 @@ impl<'ctx> InstructionValue<'ctx> {
 
     // SubTypes: Only apply to memory access instructions
     /// Returns whether or not a memory access instruction is volatile.
+    #[llvm_versions(3.6..=9.0)]
     pub fn get_volatile(self) -> Result<bool, &'static str> {
         // Although cmpxchg and atomicrmw can have volatile, LLVM's C API
-        // does not export that functionality.
+        // does not export that functionality until 10.0.
         if !self.is_a_load_inst() && !self.is_a_store_inst() {
             return Err("Value is not a load or store.");
         }
         Ok(unsafe { LLVMGetVolatile(self.as_value_ref()) } == 1)
     }
 
+    // SubTypes: Only apply to memory access instructions
+    /// Returns whether or not a memory access instruction is volatile.
+    #[llvm_versions(10.0..=latest)]
+    pub fn get_volatile(self) -> Result<bool, &'static str> {
+        if !self.is_a_load_inst() && !self.is_a_store_inst() &&
+           !self.is_a_atomicrmw_inst() && !self.is_a_cmpxchg_inst() {
+            return Err("Value is not a load, store, atomicrmw or cmpxchg.");
+        }
+        Ok(unsafe { LLVMGetVolatile(self.as_value_ref()) } == 1)
+    }
+
     // SubTypes: Only apply to memory access instructions
     /// Sets whether or not a memory access instruction is volatile.
+    #[llvm_versions(3.6..=9.0)]
     pub fn set_volatile(self, volatile: bool) -> Result<(), &'static str> {
         // Although cmpxchg and atomicrmw can have volatile, LLVM's C API
-        // does not export that functionality.
+        // does not export that functionality until 10.0.
         if !self.is_a_load_inst() && !self.is_a_store_inst() {
             return Err("Value is not a load or store.");
         }
         Ok(unsafe { LLVMSetVolatile(self.as_value_ref(), volatile as i32) })
     }
+    
+    // SubTypes: Only apply to memory access instructions
+    /// Sets whether or not a memory access instruction is volatile.
+    #[llvm_versions(10.0..=latest)]
```

Comment:
I wonder if, rather than duplicating these functions we can just simplify like so:

```Rust
// Comment here why this is necessary
#[llvm_versions(3.6..=9.0)]
let not_atomicrmw_cmpxchg = true;
#[llvm_version(10.0..=latest)]
let not_atomicrmw_cmpxchg = !self.is_a_atomicrmw_inst() && !self.is_a_cmpxchg_inst();

if !self.is_a_load_inst() && !self.is_a_store_inst() && not_atomicrmw_cmpxchg {
```

Reply:
I tried that and got this error:
```
error[E0658]: custom attributes cannot be applied to statements
   --> src/values/instruction_value.rs:244:9
    |
244 |         #[llvm_versions(10.0..=latest)]
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #54727 <https://github.com/rust-lang/rust/issues/54727> for more information
    = help: add `#![feature(proc_macro_hygiene)]` to the crate attributes to enable
```

I tried a couple implementations, one putting it above each `let` statement as you suggest, and another factoring out the let part and putting the feature test on the block that it's equal to. I couldn't get it to work.

Reply:
Weird, I thought they worked on statements. I think support for them on exprs is coming in a couple of versions. But I guess this is fine for now.